Improve level detection, allow disabling in specific TreeSitter classes, add toggle functionality and lazy-loading

savchenko opened the pull request:
### Features

#### Lazy-loading
Works, tested with `Lazy.nvim`

#### Indent level detection
Hack-ish, but works. The `line_text:find()` seems to be unavoidable.

#### Disable in docstrings
Works, unsurprisingly relies on TS.

#### Enable/Disable/Toggle
Works reliably.

### Changes

- `exclude_nodetype` config option which defaults to `{ 'string', 'comment' }` 
- `enabled` config option to control the default state.
- `key` option to control the toggle.
- Optional `vim.g.indentmini_key` to define a toggle hotkey.
- Massaged the Readme to make it slightly more presentable.

### Issues

Close https://github.com/nvimdev/indentmini.nvim/issues/36
Close https://github.com/nvimdev/indentmini.nvim/issues/39
Close https://github.com/nvimdev/indentmini.nvim/issues/20

savchenko committed:
fix: Indent level detection

Close https://github.com/nvimdev/indentmini.nvim/issues/36

savchenko committed:
feat: Allow disabling in (doc)strings

Close https://github.com/nvimdev/indentmini.nvim/issues/39

savchenko committed:
feat: Add disable/enable/toggle and default state

- User commands: IndentEnable, IndentDisable, IndentToggle
- API: .toggle(), .disable(), .toggle()
- Optional hotkey: `key` in the config{}
- Default state: `enabled` in the config{}

Close https://github.com/nvimdev/indentmini.nvim/issues/20

savchenko committed:
Improve README.md

Just make it more readable

savchenko commented:
@glepnir , I have little experience with automated Lua testing, but verified it manually. 
Everything seems to work, including the recently added toggle; please have a look.

savchenko committed:
Update README

Updated README to improve formatting and clarity.

glepnir commented on lua/indentmini/init.lua:
```diff
@@ -279,16 +295,70 @@ local function on_win(_, winid, bufnr, toprow, botrow)
+
+function M.enable()
+  enabled = true
+  vim.cmd('redraw!')
```
can we use `api.nvim__redraw` instead ?

savchenko replied:
Done in 534dc9baf08ef182276157002ac0366fd4aee6d3

savchenko committed:
Replace .cmd('redraw!') with .api.nvim__redraw()

savchenko committed:
feat: Support lazy-load

savchenko committed:
fix: Hotkeys when lazy-loading

We need to either:
- Rely on a global variable
- Tell plugin manager on which condition to lazy-load it

savchenko commented:
(I propose to tag a release once this is merged)

glepnir commented:
thanks !!

glepnir commented:
Hi seems like something is broken in comment line, file is `vim/src/mark.c` 🤔

<img width="1774" height="1278" alt="image" src="https://github.com/user-attachments/assets/323dc783-6a87-4088-8da8-75af54288d0c" />

savchenko commented:
@glepnir, assuming that we're looking at the same file version, `vi:set ts=8 sts=4 sw=4 noet:` might be to blame. Reindent the file (`=`)

<img width="788" height="519" alt="image" src="https://github.com/user-attachments/assets/694c7aa6-9b15-498c-887d-f0b06e82d8aa" />

glepnir commented:
problem is the 415 416 no indent line.  I use `only_current = true` . yes vim has special mode line  set for indent

glepnir commented:
another broken ..

<img width="1322" height="1382" alt="image" src="https://github.com/user-attachments/assets/4154c6f1-7d48-48af-b3e1-36a5d536724a" />

savchenko commented:
Repro?

<img width="1265" height="278" alt="image" src="https://github.com/user-attachments/assets/7d8306ae-a777-4eba-8107-59a38f77ca75" />

<img width="903" height="407" alt="image" src="https://github.com/user-attachments/assets/2ae360e1-8934-468d-b8e9-d44586157b7c" />

Two screenshots above are made using 0.11.5

glepnir commented:
use `only_current = true` 🤔 
file is `TEST_FILE=test/functional/editor/mark_spec.lua`
then move cursor to line 474
